### PR TITLE
[DF] Introduce RDatasetSpec and use entry range in RLoopManager

### PR DESCRIPTION
**Pull request by ikabadzhov**
- [x] Test the behaviour of RDatasetSpec
- [ ] Apply RDatasetSpec in the tutorials

### CODE REVIEW

File: tree/dataframe/test/dataframe_datasetspec.cxx [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+
+   // "reference" values
+   const auto dfSimple = RDataFrame("tree", "file.root").Display<int>({"x"})->AsString();
+   const auto dfRange0 = RDataFrame("tree", "file.root").Range(2, 4).Display<int>({"x"})->AsString();
+
+   // specify only tree and file names
+   const auto dfRDS0 = RDataFrame(RDatasetSpec("tree", "file.root")).Display<int>({"x"})->AsString();
+   EXPECT_EQ(dfRDS0, dfSimple);
+
+   // specify tree name, file name, starting index => [2, 0) => range is disregarded (all entries are displayed)
```
Comment (vepadulano) [inline tree/dataframe/test/dataframe_datasetspec.cxx:24]: This is not intented, the range of entries should either be specified or not

File: tree/dataframe/test/dataframe_datasetspec.cxx [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+                              .Display<int>({"x"})
+                              ->AsString(),
+                           "TChain::LoadTree", "Cannot find tree with name nottree in file file.root");
+      } catch (const std::runtime_error &err) {
+         EXPECT_EQ(std::string(err.what()),
+                   "Column \"x\" is not in a dataset and is not a custom column been defined.");
+         throw;
+      },
+      std::runtime_error);
+
```
Comment (vepadulano) [inline tree/dataframe/test/dataframe_datasetspec.cxx:69]: Also this is not intended behaviour

Comment (ikabadzhov) [reply tree/dataframe/test/dataframe_datasetspec.cxx:69]: So, do we expect only 1 ROOT error, saying: `"<TChain::LoadTree> Cannot find tree with name nottree in file file.root"`?

File: tree/dataframe/test/dataframe_datasetspec.cxx (hunk 1 above)
Comment (ikabadzhov) [reply tree/dataframe/test/dataframe_datasetspec.cxx:24]: Just to make sure, is the remark towards 1.`RDatasetSpec("tree", "file.root")` or 2.`RDatasetSpec("tree", "file.root", 2)`.

In either case, I have a reason to test that:
* If it is 1 => as you said the range could be specified or not, and here it is not specified
* If it is 2 => maybe a user tries to use this feature similarly to applying `Range(2)`, which is not allowed. In addition, I am testing what would happend if 2 valid indeces are entered, but the starting index is bigger than the ending one.

Comment (vepadulano) [reply tree/dataframe/test/dataframe_datasetspec.cxx:24]: Yeah it is for 2

File: tree/dataframe/test/dataframe_datasetspec.cxx (hunk 2 above)
Comment (vepadulano) [reply tree/dataframe/test/dataframe_datasetspec.cxx:69]: I don't know what the issue is with github. I commented the part where you construct the RDatasetSpec but then it is visualised much below, seemingly randomly. Disregard this comment, I will just merge and upload PR to upstream

**Pull request merged**